### PR TITLE
[HttpFoundation] split FileException into specialized ones about upload handling

### PR DESCRIPTION
**Pull request by fmata**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #26411
| License       | MIT

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]

**Review by soullivaneuh — APPROVED** [review]
Looks good, thanks! :+1:

File: src/Symfony/Component/HttpFoundation/File/Exception/CantWriteFileException.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ */
+
+namespace Symfony\Component\HttpFoundation\File\Exception;
+
+/**
+ * Thrown when an error UPLOAD_ERR_CANT_WRITE occurred with UploadedFile.
+ *
+ * @author Florent Mata <florentmata@gmail.com>
+ */
+class CantWriteFileException extends FileException
```
Comment (fabpot) [inline src/Symfony/Component/HttpFoundation/File/Exception/CantWriteFileException.php]: I would nose Cant here but Cannot instead.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/HttpFoundation/File/Exception/CantWriteFileException.php (hunk 1 above)
Comment (fmata) [reply src/Symfony/Component/HttpFoundation/File/Exception/CantWriteFileException.php]: I wanted to follow the PHP const `UPLOAD_ERR_CANT_WRITE` to keep consistency, as others exceptions added in this PR. Are you sure of this change ?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpFoundation/File/Exception/CantWriteFileException.php]: @fmata can you please do the change? (and rebase meanwhile to fix the small conflict)

File: src/Symfony/Component/HttpFoundation/File/Exception/CantWriteFileException.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\File\Exception;
+
+/**
+ * Thrown when an error UPLOAD_ERR_CANT_WRITE occurred with UploadedFile.
```
Comment (xabbuh) [inline src/Symfony/Component/HttpFoundation/File/Exception/CantWriteFileException.php]: I would swap the word "error" and the name of the error.

Comment (fmata) [reply src/Symfony/Component/HttpFoundation/File/Exception/CantWriteFileException.php]: Updated

File: src/Symfony/Component/HttpFoundation/File/Exception/CantWriteFileException.php (hunk 1 above)
Comment (fmata) [reply src/Symfony/Component/HttpFoundation/File/Exception/CantWriteFileException.php]: @nicolas-grekas rebased & updated

**Comment by fmata** [comment]
Status: Needs Review

**Review by lyrixx — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @fmata.

**Pull request merged**